src/Compilers/CSharp/Portable/Generated/CSharpSyntaxGenerator/CSharpSyntaxGenerator.SourceGenerator/Syntax.xml.Main.Generated.cs

```diff
@@ -4807,13 +4798,20 @@ public static InterfaceDeclarationSyntax InterfaceDeclaration(string identifier)
             => SyntaxFactory.InterfaceDeclaration(default, default(SyntaxTokenList), SyntaxFactory.Token(SyntaxKind.InterfaceKeyword), SyntaxFactory.Identifier(identifier), default, default, default, SyntaxFactory.Token(SyntaxKind.OpenBraceToken), default, SyntaxFactory.Token(SyntaxKind.CloseBraceToken), default);
 
         /// <summary>Creates a new RecordDeclarationSyntax instance.</summary>
-        public static RecordDeclarationSyntax RecordDeclaration(SyntaxList<AttributeListSyntax> attributeLists, SyntaxTokenList modifiers, SyntaxToken keyword, SyntaxToken classKeyword, SyntaxToken identifier, TypeParameterListSyntax? typeParameterList, ParameterListSyntax? parameterList, BaseListSyntax? baseList, SyntaxList<TypeParameterConstraintClauseSyntax> constraintClauses, SyntaxToken openBraceToken, SyntaxList<MemberDeclarationSyntax> members, SyntaxToken closeBraceToken, SyntaxToken semicolonToken)
+        public static RecordDeclarationSyntax RecordDeclaration(SyntaxKind kind, SyntaxList<AttributeListSyntax> attributeLists, SyntaxTokenList modifiers, SyntaxToken keyword, SyntaxToken classOrStructKeyword, SyntaxToken identifier, TypeParameterListSyntax? typeParameterList, ParameterListSyntax? parameterList, BaseListSyntax? baseList, SyntaxList<TypeParameterConstraintClauseSyntax> constraintClauses, SyntaxToken openBraceToken, SyntaxList<MemberDeclarationSyntax> members, SyntaxToken closeBraceToken, SyntaxToken semicolonToken)
```

Comment:
Is there a backcompat overload?

Reply:
There is (in `RecordDeclarationSyntax.cs`). CI cannot pass if we regress the shipped public API.

Reply:
@jcouv I don't see any overload in `RecordDeclarationSyntax.cs` has a `SyntaxToken classKeyword` parameter. I might be missing something.

> CI cannot pass if we regress the shipped public API.

There is an open bug around source generators and public API analyzer: https://github.com/dotnet/roslyn-analyzers/issues/4729. That's why I attempted to manually verify.

Reply:
This overload is actually unshipped. So no need for backcompat for it :smile: Marking the conversation as resolved.

Reply:
> There is an open bug around source generators and public API analyzer: dotnet/roslyn-analyzers#4729. That's why I attempted to manually verify.

Interesting. From my experience in recent months, publicAPI analyzer has been reliable from the command-line (`build -runAnalyzers`) as used in CI, but diagnostics were not reported in the IDE.